### PR TITLE
Update macOS Prerequisites

### PR DESCRIPTION
**Pull request by eerhardt**
Updating macOS prerequisites to the latest dependencies

### CODE REVIEW

File: docs/building/unix-instructions.md [hunk 1]
```diff
@@ -42,11 +42,12 @@ macOS 10.12 (Sierra) or higher is needed to build dotnet/machinelearning.
 
 On macOS a few components are needed which are not provided by a default developer setup:
 * cmake 3.10.3
-* gcc
+* libomp
+* libgdiplus
+* gettext
```
Comment (Ivanidzo4ka) [inline docs/building/unix-instructions.md]: libomp for MKL I assume, libgdiplus for System.Drawing, what gettext for?

**Comment by TomFinley** [comment]
Our of curiosity, we used to need `gcc` I think for LightGBM since otherwise the package would not work, but I see we've explicitly deleted that -- has that issue been resolved somehow?

**Comment by eerhardt** [comment]
> Our of curiosity, we used to need gcc I think for LightGBM since otherwise the package would not work, but I see we've explicitly deleted that -- has that issue been resolved somehow?

Yes, the latest LightGBM for macOS no longer explicitly requires `gcc`. Instead, it just requires `libomp`.

See https://github.com/Microsoft/LightGBM/issues/1523
https://github.com/Microsoft/LightGBM/pull/1676/files#diff-f54ede2a43d00dde2f6ff27364db0e9dR67

Note: this command is the exact command that our PR validation builds use:

https://github.com/dotnet/machinelearning/blob/6c5677a3fd8d6f2825cc5b14b29b81653ba9d7c0/build/ci/phase-template.yml#L28

And they run on "normal" macOS machines.

File: docs/building/unix-instructions.md (hunk 1 above)
Comment (eerhardt) [reply docs/building/unix-instructions.md]: `libomp` is for `LightGBM`.

`libgdiplus` is for System.Drawing. I guess `gettext` is for `libgdiplus`: https://github.com/mono/libgdiplus/blob/f1dfa1bf87b26dfe74448a3fff9716793a78e98e/README.md#requirements

**Pull request merged**